### PR TITLE
ctp: add foundry tests for Drippie

### PR DESCRIPTION
**Pull request by tynes**
**Description**

This commit adds foundry tests for the `Drippie` smart contract. Test coverage is improved as additional events that were not tested before are now being tested for. Also allows a future version of `Drippie` to be more easily invariant tested.

All hardhat tests are ported 1:1 with additional assertion of specific `CALL`s and events being emitted.

### CODE REVIEW

File: packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+        vm.prank(owner);
+
+        vm.expectEmit(true, true, true, true);
+        emit DripStatusUpdated({
+            nameref: dripName,
+            name: dripName,
+            status: Drippie.DripStatus.ACTIVE
+        });
+
+        drippie.status(dripName, Drippie.DripStatus.ACTIVE);
```
Comment (refcell) [inline packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol:226]: I wish this method was `setStatus` as opposed to `status` to make it's state transition explicit.

File: packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+            abi.encodeWithSelector(SimpleStorage.set.selector, keyOne, valueOne)
+        );
+
+        vm.expectCall(
+            address(simpleStorage),
+            0,
+            abi.encodeWithSelector(SimpleStorage.set.selector, keyTwo, valueTwo)
+        );
+
+        drippie.drip(dripName);
```
Comment (refcell) [inline packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol:495]: ibid

**Review by refcell — CHANGES_REQUESTED** [review]
Minor nits - also would be nice to test reentrancy within one of the e2e tests.

File: packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol (hunk 1 above)
Comment (roninjin10) [reply packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol:226]: +1 it looks a lot like a getter not a setter

File: packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+        vm.stopPrank();
+    }
+
+    /**
+     * @notice Ensures that only the owner of Drippie can create a drip.
+     */
+    function test_fails_unauthorized() external {
+        vm.expectRevert("UNAUTHORIZED");
+        drippie.create(dripName, _defaultConfig());
+    }
```
[the remark this answers is not in the document]
Comment (roninjin10) [reply packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol:200]: I'm a forge noob.  @refcell so by naming a param fuzzFoo foundry knows to run a fuzz test here?   Thata's all one has to do to foundry test?!?!

File: packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol (hunk 1 above)
Comment (tynes) [reply packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol:226]: Unfortunately we can't change the abi right now and drippie v1 works well enough for now, there are some concepts for v2 out there but no pressing need to implement right now besides for fun

File: packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (tynes) [reply packages/contracts-periphery/contracts/foundry-tests/Drippie.t.sol:200]: Yes to fuzz all you need to do is pass params to a function with the `test` prefix

**Comment by tynes** [comment]
I've addressed all comments on this PR as well as rebased it on latest `develop`

cc @refcell

**Review by refcell — APPROVED** [review]
💯

**Pull request merged**